src/main/java/org/elasticsearch/cluster/action/index/MappingUpdatedAction.java

```diff
@@ -134,179 +118,9 @@ public void updateMappingOnMasterSynchronously(String index, String indexUUID, S
-        private CompressedString mappingSource;
-        private String nodeId = null; // null means not set
-
-        MappingUpdatedRequest() {
-        }
-
-        public MappingUpdatedRequest(String index, String indexUUID, String type, CompressedString mappingSource, String nodeId) {
-            this.index = index;
-            this.indexUUID = indexUUID;
-            this.type = type;
-            this.mappingSource = mappingSource;
-            this.nodeId = nodeId;
-        }
-
-        public String index() {
-            return index;
-        }
-
-        @Override
-        public IndicesOptions indicesOptions() {
-            return IndicesOptions.strictSingleIndexNoExpandForbidClosed();
-        }
-
-        @Override
-        public String[] indices() {
-            return new String[]{index};
-        }
-
-        public String indexUUID() {
-            return indexUUID;
-        }
-
-        public String type() {
-            return type;
-        }
-
-        public CompressedString mappingSource() {
-            return mappingSource;
-        }
-
-        /**
-         * Returns null for not set.
-         */
-        public String nodeId() {
-            return this.nodeId;
-        }
-
-        @Override
-        public ActionRequestValidationException validate() {
-            return null;
-        }
-
-        @Override
-        public void readFrom(StreamInput in) throws IOException {
-            super.readFrom(in);
-            index = in.readString();
-            type = in.readString();
-            mappingSource = CompressedString.readCompressedString(in);
-            indexUUID = in.readString();
-            nodeId = in.readOptionalString();
-        }
-
-        @Override
-        public void writeTo(StreamOutput out) throws IOException {
-            super.writeTo(out);
-            out.writeString(index);
-            out.writeString(type);
-            mappingSource.writeTo(out);
-            out.writeString(indexUUID);
-            out.writeOptionalString(nodeId);
-        }
-
-        @Override
-        public String toString() {
-            return "index [" + index + "], indexUUID [" + indexUUID + "], type [" + type + "] and source [" + mappingSource + "]";
-        }
-    }
-
-    private static class MappingChange {
-        public final String index;
-        public final String indexUUID;
-        public final String type;
-        public final CompressedString mappingSource;
-        public final MappingUpdateListener listener;
-
-        MappingChange(String index, String indexUUID, String type, CompressedString mappingSource, MappingUpdateListener listener) {
-            this.index = index;
-            this.indexUUID = indexUUID;
-            this.type = type;
-            this.mappingSource = mappingSource;
-            this.listener = listener;
+    public void updateMappingOnMasterSynchronously(String index, String type, Mapping mappingUpdate, TimeValue timeout) throws Throwable {
```

Comment:
so I wonder if we should pass it on though. I think it's good to validate it and if we use the client as a user of ES we just pass `IndexMetaData#INDEX_UUID_NA_VALUE` as the default?